`==` can't match `.init` on Xcode 12.5

- [ x] I have read [CONTRIBUTING](https://github.com/Quick/Nimble/blob/master/CONTRIBUTING.md) and have done my best to follow them.

### What did you do?

```
struct Xxx: Equatable {
    let value: Int
}
let xxx = Xxx(value: 123)
expect(xxx) == .init(value: 123)
```

### What did you expect to happen?

Successfuly compile and pass expect, as it used to with XCode 12.4 & Swift 5.3.

### What actually happened instead?

```
/Tests.swift:128:25: error: binary operator '==' cannot be applied to operands of type 'Expectation<Xxx>' and '_'
            expect(xxx) == .init(value: 123)
            ~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~
/Tests.swift:128:29: error: cannot infer contextual base in reference to member 'init'
            expect(xxx) == .init(value: 123)
                           ~^~~~
```

### Environment

List the software versions you're using:

- Nimble: 9.0.1
- Xcode Version: Version 12.5 (12E262)
- Swift Version: Xcode Default

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

- Cocoapods: 1.10.1


### Project that demonstrates the issue

https://github.com/serges147/nimble_report

--- a/Sources/Nimble/Matchers/Equal.swift
+++ b/Sources/Nimble/Matchers/Equal.swift
@@ -102,8 +102,16 @@ private func equal<T>(_ expectedValue: Set<T>?, stringify: @escaping (Set<T>?) -
     }
 }
 
+public func ==<T: Equatable>(lhs: Expectation<T>, rhs: T) {
+    lhs.to(equal(rhs))
+}
+
 public func ==<T: Equatable>(lhs: Expectation<T>, rhs: T?) {
     lhs.to(equal(rhs))
+}
+
+public func !=<T: Equatable>(lhs: Expectation<T>, rhs: T) {
+    lhs.toNot(equal(rhs))
 }
 
 public func !=<T: Equatable>(lhs: Expectation<T>, rhs: T?) {
@@ -118,16 +126,32 @@ public func !=<T: Equatable>(lhs: Expectation<[T]>, rhs: [T]?) {
     lhs.toNot(equal(rhs))
 }
 
+public func == <T>(lhs: Expectation<Set<T>>, rhs: Set<T>) {
+    lhs.to(equal(rhs))
+}
+
 public func == <T>(lhs: Expectation<Set<T>>, rhs: Set<T>?) {
     lhs.to(equal(rhs))
+}
+
+public func != <T>(lhs: Expectation<Set<T>>, rhs: Set<T>) {
+    lhs.toNot(equal(rhs))
 }
 
 public func != <T>(lhs: Expectation<Set<T>>, rhs: Set<T>?) {
     lhs.toNot(equal(rhs))
 }
 
+public func ==<T: Comparable>(lhs: Expectation<Set<T>>, rhs: Set<T>) {
+    lhs.to(equal(rhs))
+}
+
 public func ==<T: Comparable>(lhs: Expectation<Set<T>>, rhs: Set<T>?) {
     lhs.to(equal(rhs))
+}
+
+public func !=<T: Comparable>(lhs: Expectation<Set<T>>, rhs: Set<T>) {
+    lhs.toNot(equal(rhs))
 }
 
 public func !=<T: Comparable>(lhs: Expectation<Set<T>>, rhs: Set<T>?) {

--- a/Tests/NimbleTests/Matchers/EqualTest.swift
+++ b/Tests/NimbleTests/Matchers/EqualTest.swift
@@ -158,6 +158,17 @@ final class EqualTest: XCTestCase {
         }
     }
 
+    // see: https://github.com/Quick/Nimble/issues/867
+    func testOperatorEqualityWithImplicitMemberSyntax() {
+        struct Xxx: Equatable {
+            let value: Int
+        }
+
+        let xxx = Xxx(value: 123)
+        expect(xxx) == .init(value: 123)
+        expect(xxx) != .init(value: 124)
+    }
+
     func testOperatorEqualityWithArrays() {
         let array1: [Int] = [1, 2, 3]
         let array2: [Int] = [1, 2, 3]

